src/back/spv/writer.rs

```diff
@@ -1019,6 +983,29 @@ impl Writer {
         Ok(id)
     }
 
+    fn get_index_constant(
+        &mut self,
+        index: Word,
+        types: &Arena<crate::Type>,
+    ) -> Result<Word, Error> {
+        self.get_constant_scalar(crate::ScalarValue::Uint(index as _), 4, types)
+    }
```

Comment:
This changes the types of indexes used as LOD selectors and `OpAccessChain` operands from `Sint` to `Uint`. I couldn't figure out the rules for LOD selection, but `OpAccessChain` says the indices are "treated as signed", so is this what we want?

Reply:
SPIR-V doesn't care, basically. It will reinterpret them as signed, and it's still fine.